elearning-frontend/src/components/ProfileSettings.js

```diff
@@ -0,0 +1,228 @@
+import React, { useState, useEffect, useRef } from 'react'
+import apiClient from '../services/api';
+import { useNavigate } from 'react-router-dom';
+
+const ProfileSettings = (props) => {
+
+    const navigate = useNavigate();
+
+    // User Details
+    const [profilepic, setProfilepic] = useState('https://st3.depositphotos.com/4111759/13425/v/600/depositphotos_134255710-stock-illustration-avatar-vector-male-profile-gray.jpg');
+    const [avatar, setAvatar] = useState(null);
+    const avatarRef = useRef(null);
+    const [firstname, setFirstname] = useState('');
+    const [middlename, setMiddlename] = useState('');
+    const [lastname, setLastname] = useState('');
+    const [sex, setSex] = useState('');
+    const [birthdate, setBirthdate] = useState('');
+    const [phone, setPhone] = useState('');
+    const [address, setAddress] = useState('');
+    const [email, setEmail] = useState('');
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+
+    const [loading, setLoading] = useState(true);
+    const usertoken = localStorage.getItem('user');
+
+    useEffect(() => {
+        if (!props.session) {
+            navigate('/')
+        }
+        else {
+            apiClient({
+                method: "post",
+                url: "/api/user",
+                headers: {
+                    Authorization: 'Bearer ' + usertoken
+                },
+                data: {
+                    token: usertoken
+                }
+            }).then(response => {
+                
+                if (response.data.user.profile.avatar !== null && response.data.user.profile.avatar != "") {
+                    setAvatar(response.data.user.profile.avatar);
+                    setProfilepic('http://127.0.0.1:8000/uploads/avatar/' + response.data.user.profile.avatar);
+                }
+
+                setFirstname(response.data.user.profile.firstname);
+                setMiddlename(response.data.user.profile.middlename);
+                setLastname(response.data.user.profile.lastname);
+                setSex(response.data.user.profile.sex);
+                setBirthdate(response.data.user.profile.birthdate);
+                setPhone(response.data.user.profile.phone);
+                setAddress(response.data.user.profile.address);
+                setEmail(response.data.user.email);
+                setUsername(response.data.user.username);
+                setPassword(response.data.user.password);
```

Comment:
instead of creating state of every data. You use this instead
```
const [profile, setProfile] = useState({
//initial values here
profile_image: 'link here',
name: 'name'
})
```

then in your api response
you can use this one

`setProfile(response.data)`

how you access your data:
```
{profile.name}
{profile.email}
...
```